packages/tools/src/tools/FreehandROISculptorTool.ts

```diff
@@ -0,0 +1,685 @@
+        referencedToolName: 'PlanarFreehandROI',
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+    this._endCallback = this._endCallback.bind(this);
+    this._dragCallback = this._dragCallback.bind(this);
+  }
+
+  preMouseDownCallback = (evt: EventTypes.InteractionEventType) => {
+    const eventData = evt.detail;
+    const element = eventData.element;
+
+    this._configureToolSize(evt);
+    this._selectFreehandTool(eventData);
+
+    if (this._commonData.selectedIndex === null) {
+      return;
+    }
+
+    this.isActive = true;
+
+    hideElementCursor(element);
+    this._activateSculpt(element);
+    return true;
+  };
+
+  mouseMoveCallback = (evt: EventTypes.InteractionEventType): void => {
+    if (this.mode === ToolModes.Active) {
+      this._configureToolSize(evt);
+      this._updateCursor(evt);
+    } else {
+      this._commonData.mouseLocation = undefined;
+    }
+  };
+
+  /**
+   * Updates cursor size
+   *
+   * @param evt - The event
+   */
+  _updateCursor(evt: EventTypes.InteractionEventType): void {
+    const eventData = evt.detail;
+    const element = eventData.element;
+    const config = this.configuration;
+
+    const enabledElement = getEnabledElement(element);
+    const { renderingEngine, viewport } = enabledElement;
+
+    this._commonData.viewportIdsToRender = [viewport.id];
+
+    const annotations =
+      this._filterInteractableFreehandRoiAnnotationsForElement(element);
+
+    if (!annotations?.length) {
+      return;
+    }
+
+    this._commonData.mouseLocation = eventData.currentPoints.canvas;
+
+    if (this.isActive) {
+      annotations[this._commonData.selectedIndex].highlighted = true;
+    } else {
+      const canvasCoords = eventData.currentPoints.canvas;
+      const radius = this._distanceFromPoint(
+        viewport,
+        annotations[this._commonData.selectedIndex]?.data,
+        canvasCoords
+      );
+      if (radius > 0) {
+        this.toolSize = Math.min(config.maxToolSize, radius);
+      }
+    }
+
+    triggerAnnotationRenderForViewportIds(
+      renderingEngine,
+      this._commonData.viewportIdsToRender
+    );
+  }
+
+  /**
+   * Returns interactable freehand ROI annotations
+   *
+   * @param element - The viewport element
+   */
+  _filterInteractableFreehandRoiAnnotationsForElement(element: HTMLDivElement) {
+    const config = this.configuration;
+    const enabledElement = getEnabledElement(element);
+    const { renderingEngineId, viewportId } = enabledElement;
+    let interactableAnnotation = [];
+
+    const toolGroup = ToolGroupManager.getToolGroupForViewport(
+      viewportId,
+      renderingEngineId
+    );
+
+    const toolInstance = toolGroup.getToolInstance(config.referencedToolName);
+
+    config.referencedToolNames.forEach((referencedToolName: string) => {
+      const annotation = getAnnotations(referencedToolName, element);
+      if (annotation) {
+        interactableAnnotation = [...interactableAnnotation, ...annotation];
```

Comment:
Could you return Map<annotationUID,ContourAnnotation> here please?  That should be declared in the return type.

Reply:
Below function `filterInteractableAnnotationsForElement` and this function usages expect annotations to be in array format. Changing this expectation would result in unnecessary conversions between map and array data structures.